mosaic/datasets.py

```diff
@@ -0,0 +1,91 @@
+import pathlib
+
+import pooch
+import xarray as xr
+from xarray import Dataset
+
+from mosaic.version import __version__
+
+# this dictionary uses the short handed mesh names for convience,
+# but can not be parsed by pooch
+registry = {
+    "QU.960km": {
+        "lcrc_path": "mpas_standalonedata/mpas-ocean/mesh_database/mesh.QU.960km.151026.nc",
+        "sha256_hash": "sha256:524d2d5a93851395a3bdfafb30a5acb2a3dbbecd0db07cd29e5e3f87da6eb82f"
+    },
+
+    "QU.240km": {
+        "lcrc_path": "inputdata/ocn/mpas-o/oQU240/ocean.QU.240km.151209.nc",
+        "sha256_hash": "sha256:a3758f88ceff3d91e86dba7922f6dd7d5672157b4793ef78214624ab8b2724ae"
+    },
+
+    "mpaso.EC30to60E2r3": {
+        "lcrc_path": "inputdata/ocn/mpas-o/EC30to60E2r3/mpaso.EC30to60E2r3.230313.nc",
+        "sha256_hash": "sha256:55e7cc33c890f7b9f1188bcc07fd8218b57cb1cd5ba32ee66fe3162a36995a7c"
+    },
+
+    "mpasli.AIS8to30": {
+        "lcrc_path": "inputdata/glc/mpasli/mpas.ais8to30km/ais_8to30km.20221027.nc",
+        "sha256_hash": "sha256:932a1989ff8e51223413ef3ff0056d6737a1fc7f53e440359884a567a93413d2"
+    }
+}
+
+# create a parsable registry for pooch from human friendly one
+_registry = {registry[m]["lcrc_path"]: registry[m]["sha256_hash"] for m in registry}
+
+mesh_db = pooch.create(
+    # Use the default cache folder for the operating system
+    path=pooch.os_cache("mosaic"),
```

Comment:
This location may be appropriate for small files but maybe not for the EC30to60 mesh, which I think is too big to cache in someone's home directory.

Reply:
Maybe there's not a great way around this.  But is there some way a user can specify a different path?

Reply:
It looks like this would work:
```
from mosaic.datasets import mesh_db

mesh_db.path = '/lcrc/group/e3sm/ac.xylar/mosaic_cache'
```
Maybe add that to the docs?

Reply:
The hope was to avoid users having to fiddle with the `mesh_db` object, which is why there's the wrapper `open_dataset` function. 

In the most recent commit [e7d28b2](https://github.com/E3SM-Project/mosaic/pull/9/commits/e7d28b2ceff3ae6de8f4bc2b19fd8af06e05b04e), I've added a `cache_dir` key word argument to `open_datasets` function which adds this support for custom cache location.

Maybe it's make sense to add a note to the documentation that the cached location will be in your home directory on linux system, which could cause problems on most HPC platforms. Sounds like we can use [environmental variable to set the cache location](https://www.fatiando.org/pooch/latest/user-defined-cache.html), so we could explore setting the default cache location per system (probably not needed of now at least).

Reply:
Okay, great.  Both the parameter and the environment variable sound like good ways to handle this if folks have trouble with the cache files taking up too much space.